docker/Dockerfile

```diff
@@ -0,0 +1,62 @@
+# docker buildx build --push --platform linux/amd64,linux/arm64/v8 -o type=image -t briansimulator/brian -f docker/Dockerfile .
+# docker run -it --init --rm -p 8888:8888 briansimulator/brian
+
+ARG BASE_IMAGE_TAG=3.12-slim-bookworm
+FROM python:${BASE_IMAGE_TAG}
+
+LABEL maintainer="Ben Evans <B.D.Evans@sussex.ac.uk>" \
+    org.opencontainers.image.vendor="The Brian Development Team" \
+    org.opencontainers.image.licenses="CeCILL 2.1" \
+    org.opencontainers.image.title="Brian Docker Image" \
+    org.opencontainers.image.description="Docker image for Brian - a free, open source simulator for spiking neural networks" \
+    org.opencontainers.image.homepage="https://briansimulator.org" \
+    org.opencontainers.image.url="https://hub.docker.com/r/briansimulator/brian" \
+    org.opencontainers.image.source="https://github.com/brian-team/brian2" \
+    org.opencontainers.image.documentation="https://brian2.readthedocs.io/"
+
+# Install dependencies
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    build-essential \
+    g++ \
+    git \
+    sudo \
+    && rm -rf /var/lib/apt/lists/*
+
+# Copy wheels into the image
+COPY dist /tmp/dist
+
+# Install Brian2 and recommended packages
+ARG TARGETARCH
```

Comment:
Would it be preferable to get the architecture from the image directly with something like: `dpkg --print-architecture`?

Reply:
Just to make clear that there is no misunderstanding: this is not an argument that you need to set, it is coming from docker (https://docs.docker.com/build/building/variables/#multi-platform-build-arguments). I understood that I have to declare them like this (but I didn't actually test it):
> These arguments are useful for doing cross-compilation in multi-platform builds. They're available in the global scope of the Dockerfile, but they aren't automatically inherited by build stages. To use them inside stage, you must declare them:

But there might be another way that is easier – ideally something that uses the same terminology as the pip wheel file names…

Reply:
Fair enough - six of one, half a dozen of the other I guess 🤷‍♂️